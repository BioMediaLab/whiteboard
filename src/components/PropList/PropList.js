import React from 'react'
import './PropList.css'

export default function PropList({ data, allowedProps, hideMissing }) {
  const listItems = allowedProps.map(propName => {
    return {
      key: propName,
      value: data[propName]
    }
  })

  return (
    <ul className="PropList">
      {listItems.map(item => {
        const { key, value } = item
        if (!value && hideMissing) {
          return
        }
        return (
          <li key={key}>
            <strong>{key}</strong>
            <span>{value}</span>
          </li>
        )
      })}
    </ul>
  )
}
