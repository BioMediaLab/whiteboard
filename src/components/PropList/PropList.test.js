import React from 'react'
import PropList from './'

describe('PropList', () => {
  test('can render', () => {
    const data = {
      example1: 'value',
      example2: 'value',
      example3: 'value'
    }
    const component = render(
      <PropList data={data} allowedProps={['example1']} />
    )

    expect(component).toMatchSnapshot()
  })
})
