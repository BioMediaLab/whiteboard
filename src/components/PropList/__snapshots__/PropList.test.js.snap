// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropList can render 1`] = `
<div>
  <ul
    class="PropList"
  >
    <li>
      <strong>
        example1
      </strong>
      <span>
        value
      </span>
    </li>
  </ul>
</div>
`;
